.github/workflows/end_to_end_tests.yml

```diff
@@ -519,3 +538,43 @@ jobs:
           status: ${{ job.status }}
           steps: ${{ toJson(steps) }}
           channel: '#cloud-tech'
+
+  # Runs agent from it source and performs some basic sanity checks.
+  agent-source-tests:
+    runs-on: ${{ matrix.os }}
+    needs: pre_job
+    if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
+    container: ${{ matrix.container }}
+    strategy:
+      matrix:
+        os: [ "macos-11", "ubuntu-18.04" ]
+        python-version: [ "3.7.13" ]
+        # Uncomment when windows platform and service related issues are fixed.
+    #        include:
+    #          - os: "windows-2019"
+    #            python-version: "3.7.9"
+
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Perform the deployment of the test environment
+        uses: ./.github/actions/perform-deployment
+        with:
+          deployment-name: "test_environment"
+
+      - name: Run tests
+        shell: bash
+        run: |
+          python3 -m pytest  tests/end_to_end_tests/test_from_source/test_from_source.py \
```

Comment:
It would be good to also track and upload coverage to codecov.io for this job as well.

It's fine to implement that in a separate PR.

Reply:
Yes,  maybe create another action for this too.